spec/lib/efile/nj/nj1040_calculator_spec.rb

```diff
@@ -623,12 +623,58 @@ def over_65_birth_year
     end
   end
 
+  describe 'line 28b - other retirement income exclusion' do
+    let(:intake) { create(:state_file_nj_intake) }
+
+    context 'when taxpayer is ineligible' do
+      it 'sets line 28b to 0' do
+        allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:eligible?).and_return false
+        instance.calculate
+        expect(instance.lines[:NJ1040_LINE_28B].value).to eq(0)
+      end
+    end
+
+    context 'when taxpayer is eligible' do
+      context 'when maximum_exclusion minus 28a is greater than total_eligible_nonretirement_income' do
+        it 'sets line 28b to total_eligible_nonretirement_income' do
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:eligible?).and_return true
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:total_eligible_nonretirement_income).and_return 1_000
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:calculate_maximum_exclusion).and_return 50_000
+          allow(instance).to receive(:calculate_line_28a).and_return 0
```

Comment:
A few small notes on this test:
- [the dustiest of dust] can we rearrange these three mock statements to be in the same order as the text? First maximum exclusion, then 28a, then total eligible. It makes it much easier to read and track what's going on against the behavior described
- [dust] Could we test an edge case regarding the "greater than"? ie 50,000 minus 1,000 is A LOT greater than, might be nicer to test 50,000 - 49,999 to really show the behavior.